plugins/restful/RestfulDataProviderVariable.php

```diff
@@ -0,0 +1,306 @@
+<?php
+
+/**
+ * @file
+ * Contains \RestfulDataProviderDbQuery
+ */
+
+abstract class RestfulDataProviderVariable extends \RestfulBase implements \RestfulDataProviderInterface {
+
+  /**
+   * Constructs a RestfulDataProviderVariable object.
+   *
+   * @param array $plugin
+   *   Plugin definition.
+   * @param RestfulAuthenticationManager $auth_manager
+   *   (optional) Injected authentication manager.
+   * @param DrupalCacheInterface $cache_controller
+   *   (optional) Injected cache backend.
+   * @param string $language
+   *   (optional) The language to return items in.
+   */
+  public function __construct(array $plugin, \RestfulAuthenticationManager $auth_manager = NULL, \DrupalCacheInterface $cache_controller = NULL, $language = NULL) {
+    parent::__construct($plugin, $auth_manager, $cache_controller, $language);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function controllersInfo() {
+    return array(
+      '' => array(
+        // GET returns a list of entities.
+        \RestfulInterface::GET => 'index',
+        \RestfulInterface::HEAD => 'index',
+        // POST
+        \RestfulInterface::POST => 'setVariable',
+      ),
+      // We don't know what the ID looks like, assume that everything is the ID.
+      '^.*$' => array(
+        \RestfulInterface::GET => 'viewVariables',
+        \RestfulInterface::HEAD => 'viewVariables',
+        \RestfulInterface::PUT => 'setVariable',
+        \RestfulInterface::PATCH => 'setVariable',
+        \RestfulInterface::DELETE => 'remove',
+      ),
+    );
+  }
+
+  /**
+   * Defines default sort for variable names.
+   *
+   * By default, the array of variables returned by Drupal is already sorted
+   * in ascending order by variable name.
+   *
+   * @return array
+   *   Array keyed by the sort field, with the order ('ASC' or 'DESC') as value.
+   */
+  public function defaultSortInfo() {
+    return array();
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getVariablesForList() {
+    global $conf;
+    $variables = $conf;
+
+    $this->applyListSort($variables);
+    $this->applyListPagination($variables);
+
+    return $variables;
+  }
+
+  /**
+   * Sort the list of variables by name.
+   *
+   * @param string $a
+   *   The first variable name for the comparison.
+   * @param string $b
+   *   The second variable name for the comparison.
+   */
+  public function nameSort($a, $b) {
+    return strcasecmp($a, $b);
+  }
+
+  /**
+   * Sort the list of variables by value.
+   *
+   * If you are comparing variables with the same value, note that the PHP
+   * sorting algorithm is not stable, and their order may be changed.
+   *
+   * @param mixed $a
+   *   The first data structure for the comparison.
+   * @param mixed $b
+   *   The second data structure for the comparison.
+   */
+  public function valueSort($a, $b) {
+    return;
```

Comment:
By default, we don't do any sorting on the variable values, since their data structures can be completely different.  This method exists as a placeholder, to be overridden by the plugin author.  If they have some specific variables that they know the data structure of, and they want to sort by, then they can override this method in their plugin.  Maybe I should replace the algorithm comment with something more useful?